examples/compare-output.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+import subprocess
+import sys
+import re
+import os
+import difflib
+
+if len(sys.argv) < 5:
+    print(
+        "Usage: compare-output.py <working-dir> <path-to-ginkgo-lib> <reference-file> <full-path-to-executable> <args>..."
+    )
+    sys.exit(1)
+
+cwd = sys.argv[1]
+ginkgo_path = sys.argv[2]
+reference = open(sys.argv[3]).readlines()
+args = sys.argv[4:]
+env = os.environ.copy()
+if os.name == "nt":
+    # PATH is hopefully never empty
+    env["PATH"] += ";" + ginkgo_path
+else:
+    if "LD_LIBRARY_PATH" in env:
+        env["LD_LIBRARY_PATH"] += ":" + ginkgo_path
+    else:
+        env["LD_LIBRARY_PATH"] = ginkgo_path
+
+result = subprocess.run(
+    args=args, cwd=cwd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env
```

Comment:
you can add `text=True` or `universal_newlines=True`, then you don't need `decode` below.